src/test/test_snap_mapper.cc

```diff
@@ -558,18 +558,19 @@ class MapperVerifier {
   // must be called with lock held to protect access to
   // snap_to_hobject and hobject_to_snap
   int trim_snap(snapid_t snapid, unsigned max_count, vector<hobject_t> & out) {
-    set<hobject_t>&   hobjects = snap_to_hobject[snapid];
-    vector<hobject_t> hoids;
-    int ret = mapper->get_next_objects_to_trim(snapid, max_count, &hoids);
-    if (ret == 0) {
-      out.insert(out.end(), hoids.begin(), hoids.end());
-      for (auto &&hoid: hoids) {
+
+    set<hobject_t>& hobjects = snap_to_hobject[snapid];
+    auto hoids = mapper->get_next_objects_to_trim(snapid, max_count);
+    if (hoids.has_value()) {
+
+      out.insert(out.end(), hoids->begin(), hoids->end());
+      for (auto &&hoid: *hoids) {
```

Comment:
hoids.value()

Reply:
I'm with @ronen-fr here, using std::optional::operator* and operator std::optional::bool() seem entirely fine to me -- I certainly do it all the time.